ui-v2/app/components/consul-intention-form/index.hbs

```diff
@@ -1,123 +1,191 @@
-      <fieldset>
-        <h2>Destination</h2>
-        <label data-test-destination-element class="type-select{{if _item.error.DestinationName ' has-error'}}">
-          <span>Destination Service</span>
-          <PowerSelectWithCreate
-              @options={{_services}}
-              @searchField="Name"
-              @selected={{DestinationName}}
-              @searchPlaceholder="Type service name"
-              @buildSuggestion={{action "createNewLabel" "Use a Consul Service called '{{term}}'"}}
-              @showCreateWhen={{action "isUnique"}}
-              @onCreate={{action "change" "DestinationName"}}
-              @onChange={{action "change" "DestinationName"}} as |service|>
-                  {{#if (eq service.Name '*') }}
-                    * (All Services)
-                  {{else}}
-                    {{service.Name}}
-                  {{/if}}
-          </PowerSelectWithCreate>
-          <em>Search for an existing service, or enter any Service name.</em>
-        </label>
-  {{#if (env 'CONSUL_NSPACES_ENABLED')}}
-        <label data-test-destination-nspace class="type-select{{if _item.error.DestinationNS ' has-error'}}">
-          <span>Destination Namespace</span>
-          <PowerSelectWithCreate
-              @options={{_nspaces}}
-              @searchField="Name"
-              @selected={{DestinationNS}}
-              @searchPlaceholder="Type namespace name"
-              @buildSuggestion={{action "createNewLabel" "Use a future Consul Namespace called '{{term}}'"}}
-              @showCreateWhen={{action "isUnique"}}
-              @onCreate={{action "change" "DestinationNS"}}
-              @onChange={{action "change" "DestinationNS"}} as |nspace|>
+
+<StateChart @src={{chart}} as |State Guard Action dispatch state|>
+
+  <State @matches="removing">
+    <DataSink
+      @sink={{concat '/*/' _item.Datacenter '/intention/' _item.ID}}
+      @item={{_item}}
+      @data={{null}}
+      @onchange={{action dispatch "SUCCESS"}}
+      @onerror={{queue (action (mut error) value="error.errors.firstObject") (action dispatch "ERROR")}}
+    />
+  </State>
+
+  <State @matches="persisting">
+    <DataSink
+      @sink={{concat '/*/' _item.Datacenter '/intention/' _item.ID}}
+      @item={{_item}}
+      @onchange={{action dispatch "SUCCESS"}}
+      @onerror={{queue (action (mut error) value="error.errors.firstObject") (action dispatch "ERROR")}}
+    />
+  </State>
+
+  <State @matches="removed">
+    <Notification @after={{action ondelete}}>
+      <p data-notification role="alert" class="success notification-update">
+        <strong>Success!</strong>
+        Your intention has been deleted.
+      </p>
+    </Notification>
+  </State>
+
+  <State @matches="persisted">
+    <Notification @after={{action onchange}}>
+      <p data-notification role="alert" class="success notification-update">
+        <strong>Success!</strong>
+        Your intention has been saved.
+      </p>
+    </Notification>
```

Comment:
Our notifications here have been kept as close as possible to our previous notifications, whilst I'm not keen on this, its something we can revisit down the line so you don't have to remember to add `role="alert"` etc, and it would be nice if they didn't depend on `p`'s for containers etc.